src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -17175,5 +17184,347 @@ static void Main(List<int> list)
+                }
+                """;
+
+            CreateCompilation(source).VerifyEmitDiagnostics(
+                // (4,7): error CS7036: There is no argument given that corresponds to the required parameter 's' of 'C.C(string)'
+                // C x = [];
+                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "[]").WithArguments("s", "C.C(string)").WithLocation(4, 7)
+                );
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69521")]
+        public void OptionalParameterCtor()
+        {
+            string source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                C x = [];
+
+                class C : IEnumerable<int>
+                {
+                    public C(string s = null) { }
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
+                    public void Add(int i) { }
+                }
+                """;
+
+            CreateCompilation(source).VerifyEmitDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69521")]
+        public void MissingCtor_1()
+        {
+            string source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                abstract class AbstractCollection : IEnumerable<int>
+                {
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
+                    public void Add(int i) { }
+                }
+
+                class Program
+                {
+                    static void F(AbstractCollection c) { }
+
+                    static void Main()
+                    {
+                        F([]);
+                    }
+                }
+                """;
+
+            CreateCompilation(source).VerifyEmitDiagnostics(
+                // (17,11): error CS1503: Argument 1: cannot convert from 'collection expressions' to 'AbstractCollection'
+                //         F([]);
+                Diagnostic(ErrorCode.ERR_BadArgType, "[]").WithArguments("1", "collection expressions", "AbstractCollection").WithLocation(17, 11)
+                );
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69521")]
+        public void MissingCtor_2()
+        {
+            string source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                class NoConstructorCollection : IEnumerable<int>
+                {
+                    private NoConstructorCollection() { }
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
+                    public void Add(int i) { }
+                }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

Consider adding a test that collection is a valid target type within the collection type:
```csharp
static NoConstructorCollection _c = [1, 2, 3]; // ok
```